inference-engine/tests/functional/inference_engine/transformations/detection_output_upgrade_test.cpp

```diff
@@ -0,0 +1,212 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/detection_output_upgrade.hpp"
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <ngraph/function.hpp>
+#include <ngraph/op/util/detection_output_base.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pass/manager.hpp>
+#include <string>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "transformations/init_node_info.hpp"
+
+using namespace ngraph;
+using namespace testing;
+
+namespace {
+void create_attributes_vectors(
+    std::vector<opset1::DetectionOutput::Attributes>& attrs_v1_vector,
+    std::vector<opset8::DetectionOutput::Attributes>& attrs_v8_vector) {
+  // initialize attributes affecting shape inference
+  // others remain by default
+  for (int keep_top_k : {10, -1}) {
+    for (int top_k : {5, -1}) {
+      for (bool variance_encoded_in_target : {true, false}) {
+        for (bool share_location : {true, false}) {
+          for (bool normalized : {true, false}) {
+            opset1::DetectionOutput::Attributes attrs_v1;
+            opset8::DetectionOutput::Attributes attrs_v8;
+            attrs_v1.top_k = attrs_v8.top_k = top_k;
+            attrs_v1.keep_top_k = attrs_v8.keep_top_k = {keep_top_k};
+            attrs_v1.variance_encoded_in_target =
+                attrs_v8.variance_encoded_in_target =
+                    variance_encoded_in_target;
+            attrs_v1.share_location = attrs_v8.share_location = share_location;
+            attrs_v1.normalized = attrs_v8.normalized = normalized;
+            attrs_v1_vector.push_back(attrs_v1);
+            attrs_v8_vector.push_back(attrs_v8);
+          }
+        }
+      }
+    }
+  }
+}
+}  // namespace
+
+TEST(TransformationTests, DetectionOutput1ToDetectionOutput8) {
```

Comment:
Please use TEST_F(TransformationTestsF
And update you code style.

Reply:
I will update. Thanks!

Reply:
Done.